lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -0,0 +1,180 @@
+
+require 'sinatra/base'
+
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/errors'
+require 'ddtrace/ext/http'
+
+sinatra_vs = Gem::Version.new(Sinatra::VERSION)
+sinatra_min_vs = Gem::Version.new('1.4.0')
+if sinatra_vs < sinatra_min_vs
+  raise "sinatra version #{sinatra_vs} is not supported yet " \
```

Comment:
Then the user just have not to include `ddtrace/contrib/sinatra/tracer`. This is very different from the rails case: rails instrumentation is activated by requiring `ddtrace`, so we don't want to raise an exception if rails integration is incompatible, because the user may still want to use `ddtrace` for other components. Also note that we are not crashing, we are raising an exception (which can be handled if one wishes so).